File: gerarArquivo.ts
import * as fs from 'fs';

function gerarArquivoTexto(
  nomeArquivo: string = "arquivo_1gb.txt",
  tamanhoDesejado: number = 1_000_000_000
) {
  const blocoTexto =
    "When heads roll"+"The more the better"+"Strength Determination Merciless Forever\n";

  const tamanhoBloco = Buffer.byteLength(blocoTexto, "utf-8");
  const repeticoes = Math.floor(tamanhoDesejado / tamanhoBloco);

  const stream = fs.createWriteStream(nomeArquivo, { encoding: "utf-8" });

  console.log(`Gerando arquivo ${nomeArquivo}... Isso pode levar alguns minutos.`);

  for (let i = 0; i < repeticoes; i++) {
    if (!stream.write(blocoTexto)) {
      stream.once("drain", () => {});
    }
  }

  stream.end(() => {
    console.log(
      `Arquivo '${nomeArquivo}' gerado com sucesso! Tamanho aproximado: ${(tamanhoDesejado / (1024 ** 2)).toFixed(2)} MB`
    );
  });
}

gerarArquivoTexto();

File: gerar_arquivo.py
import os

def gerar_arquivo_texto(nome_arquivo='input_mapreduce.txt', tamanho_desejado=1_073_741_824):
    file_path = os.path.join(os.getcwd(), 'shared', 'data', nome_arquivo)
    
    bloco_texto = (
        "Este é um exemplo de texto repetitivo para preencher um arquivo grande. "
        "Você pode personalizar este conteúdo conforme quiser. "
        "Apenas certifique-se de que o tamanho total atinja cerca de 1GB.\n"
    )
    
    tamanho_bloco = len(bloco_texto.encode('utf-8'))
    repeticoes = tamanho_desejado // tamanho_bloco

    with open(file_path, 'w', encoding='utf-8') as f:
        for _ in range(repeticoes):
            f.write(bloco_texto)

    print(f"Arquivo '{nome_arquivo}' gerado com sucesso! Tamanho aproximado: {tamanho_desejado // (1024**2)} MB")

if __name__ == "__main__":
    gerar_arquivo_texto()